OpenSim/Sandbox/MatlabScripts/c3d_reading.m

```diff
@@ -1,124 +1,214 @@
+% 'firstRotation'           num applies first rotation of num (dble)
+% 'axis', 'X'               applies first rotation to axis X (string)
+% 'secondRotation'          num applies second rotation of num (dble) 
+% 'axis2', 'Y'              applies second rotation to axis Y (string)
+%
+% Example ? read WalkingData.c3d and perform a 90 degree rotation about X
+%   c3d_reading('filepath', 'C:/data/WalkingData.c3d',...
+%                 'firstRotation', '90'...
+%                 'axis', 'X')
+
+function c3d_reading(varargin)
+
+p = inputParser;
+defaultFirstRotation = 0;
+default2ndRotation = 0;
+defaultFilePath = '';
+defaultAxis = '';
+expectedAxis = {'X','Y','Z'};
+
+addOptional(p,'filepath',defaultFilePath)
+addOptional(p,'firstRotation',defaultFirstRotation,@isnumeric);
+addOptional(p,'secondRotation',default2ndRotation,@isnumeric);
+addOptional(p,'axis',defaultAxis,...
+                 @(x) any(validatestring(x,expectedAxis)));
+addOptional(p,'axis2',defaultAxis,...
+                 @(x) any(validatestring(x,expectedAxis)));
+             
+parse(p,varargin{:});
+
+axis = p.Results.axis;
+axis2 = p.Results.axis2;
+filepath = p.Results.filepath;
+rot1 = p.Results.firstRotation;
+rot2 = p.Results.secondRotation;
+
+
+%% check for file path
+if isempty(filepath)
+        [filein, pathname] = uigetfile({'*.c3d','C3D file'}, 'C3D data file...');
+        filepath = fullfile(pathname,filein);
+elseif nargin >= 1
+        if exist(filepath,'file') == 0
+            error('file does not exist')
+        end        
+end
+[path, filename, ext] = fileparts(filepath);
 
+%% import java libraries
 import org.opensim.modeling.*
 
-
-
 %% Use a c3dAdapter to turn read a c3d file
-adapter = C3DFileAdapter()
-
-tables = adapter.read('test_walking.c3d')
-
+adapter = C3DFileAdapter();
+tables = adapter.read(filepath);
 
-%% get the Markers
+%% get the marker data
 markers = tables.get('markers');
 
-markerdata = opensimTimeSeriesTableToMatlab(markers);
-markerlabels = fieldnames(markerdata);
-
-% Print the (unrotated) markers to trc file
-trcfileadapter = TRCFileAdapter();
-trcfileadapter.write(markers,'test_walking.trc');
-
-%% Get the force
-% forces = tables.get('forces');
-% forcedata = opensimVec3TableToStruct(forces);
-% markerlabels = fieldnames(forcedata);
-% 
-% % Print the (unrotated) markers to trc file
-% trcfileadapter = TRCFileAdapter();
-% trcfileadapter.write(markers,'test_walking.trc');
-
+%% get the force data
+forces = tables.get('forces');
 
 %% Define a rotation matix
-Rot = 90;          
-rotationMatrix = [1,0,0;0,cos(Rot*pi/180),-(sin(Rot*pi/180));0,sin(Rot*pi/180),cos(Rot*pi/180)];
-        
-%% Rotate marker data
+rotations = rotateCoordinateSys(axis,rot1,axis2,rot2);
+
+if isempty(rotations);
+    rotate = false;
+else
+    rotate = true;
+    rotNames = fieldnames(rotations);
+    nRots    = length(rotNames);
+end
 
-new_table = TimeSeriesTableVec3()
+%% Rotate marker and force data
+if rotate
+    for i = 1 : nRots
```

Comment:
Consider combining to make a single rotation (see L139) which will provide an `nRots` factor speedup.